lib/tree.rb

```diff
@@ -16,47 +16,126 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  def add_helper(node, key, value)
+    if node.nil?
+      node = TreeNode.new(key, value)
+    elsif key < node.key
+      node.left = add_helper(node.left, key, value)
+    elsif key > node.key
+      node.right = add_helper(node.right, key, value)
+    end
+    return node
+  end
+
+  # Time Complexity: O(1)
+  # Space Complexity: O(1)
   def add(key, value)
-    raise NotImplementedError
+    @root = add_helper(@root, key, value)
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(n)
+  # Space Complexity: O(1)
   def find(key)
```

Comment:
Remember that Binary search trees do binary search with a find, so it's O(log n) for time/space.